service/src/attestation_signers.rs

```diff
@@ -0,0 +1,162 @@
+// Copyright 2023-, GraphOps and Semiotic Labs.
+// SPDX-License-Identifier: Apache-2.0
+
+use ethereum_types::{Address, U256};
+use log::{error, info, warn};
+use native::attestation::AttestationSigner;
+use std::collections::HashMap;
+use std::sync::Arc;
+use tokio::sync::RwLock;
+
+use crate::{
+    allocation_monitor::AllocationMonitor, common::allocation::allocation_signer,
+    util::create_attestation_signer,
+};
+
+#[derive(Debug, Clone)]
+pub struct AttestationSigners {
+    inner: Arc<AttestationSignersInner>,
+    _update_loop_handle: Arc<tokio::task::JoinHandle<()>>,
+}
+
+#[derive(Debug)]
+struct AttestationSignersInner {
+    attestation_signers: Arc<RwLock<HashMap<Address, AttestationSigner>>>,
+    allocation_monitor: AllocationMonitor,
+    indexer_mnemonic: String,
+    chain_id: U256,
+    dispute_manager: Address,
+}
+
+impl AttestationSigners {
+    pub fn new(
+        allocation_monitor: AllocationMonitor,
+        indexer_mnemonic: String,
+        chain_id: U256,
+        dispute_manager: Address,
+    ) -> Self {
+        let inner = Arc::new(AttestationSignersInner {
+            attestation_signers: Arc::new(RwLock::new(HashMap::new())),
+            allocation_monitor,
+            indexer_mnemonic,
+            chain_id,
+            dispute_manager,
+        });
+
+        let _update_loop_handle = {
```

Comment:
Similar question here on why do we keep _update_loop_handle for attestation signers?